### ISSUE TITLE
It would be nice if spell checker suggested correct spellings 

### ISSUE THREAD

**Issue by AEDWIP** [ipython-contrib/jupyter_contrib_nbextensions#1234]
Any thoughts about changing the implementation so that it not only highlights misspelled words but also corrects . In the past I have used gnu ispell. There are also pure JavaScript solutions like http://www.javascriptspellcheck.com/

Kind regards 

Andy

**Comment by jcb91** [comment]
Yep, would be good, but haven't got around to it for now. The current implementation _is_ pure javascript, using [Typo.js](https://github.com/cfinke/Typo.js), which is available free for any use under the modified BSD license. Intriguingly, the unlike the library you link is neither free (for publicly-acessible sites) nor open-source, (see [javascriptspellcheck.com/Free](http://www.javascriptspellcheck.com/Free) for details), and actually _isn't pure javascript_, requiring either php or ASP on the server (see [javascriptspellcheck.com/Compatibility](http://www.javascriptspellcheck.com/Compatibility)). 

Implementing this would be reasonably easy to do for anyone with the inclination, I suspect. Given that typo.js already provides a `suggest('mispeld')` method, the remaining work would be a little jquery to display the suggestions, handle picking them etc.
